src/bin/ion/commands/generate/README.md

```diff
@@ -0,0 +1,587 @@
+3. [Running the generator](#running-the-generator)
+4. [Using the generated code in a Java project](#using-the-generated-code-in-a-java-project)
+5. [Using the generated code in a Rust project](#using-the-generated-code-in-a-rust-project)
+
+## Installing `ion-cli`
+
+### Install using `brew` (Mac only)
+
+The easiest way to install the `ion-cli` is via [Homebrew](https://brew.sh/).
+Once the `brew` command is available, run:
+
+```
+brew tap amazon-ion/ion-cli
+brew install ion-cli
+```
+
+### Install using `cargo`
+
+The `ion-cli` can also be installed by using Rust's package manager, `cargo`. If you don't already have `cargo`, you can
+install it by visiting [rustup.rs](https://rustup.rs/).
+To install `ion-cli`, run the following command:
+
+```
+cargo install ion-cli
+```
+
+## Defining a Data Model
+
+The Ion code generator supports a variety of data types, both abstract and concrete
+
+* Integers
+* Strings
+* Booleans
+* Floats
+* [Records](https://en.wikipedia.org/wiki/Record_(computer_science))
+* [Lists](https://en.wikipedia.org/wiki/List_(abstract_data_type))
+* [Enumerations](https://en.wikipedia.org/wiki/Enumerated_type)
+* Nominally distinct "[new types](https://doc.rust-lang.org/rust-by-example/generics/new_types.html)"
+
+### Defining a record type
+
+A record type is defined by a type definition with a closed `fields` constraint.
+
+**Example**
+
+```
+$ion_schema_2_0
+
+type::{
+    name: customer,
+    fields: closed::{
+      first_name: string,
+      last_name: string,
+      id: int,
+    }
+}
+```
+
+### Using Lists
+
+The Ion code generator supports homogeneous lists. To use a homogenous list as a field type, you can specify it inline
+using the `element` constraint.
+
+To define a new type that is a homogeneous list, create a top-level type with an `element` constraint.
+
+**Example**
+
+```
+$ion_schema_2_0
+
+type::{
+  name: product_tags,
+  type: list,
+  element: string,
+}
+
+type::{
+  name: product_summary,
+  fields: closed::{
+    product_id: int,
+    name: string,
+    
+    // The type of the `tags` field is the new type `product_tags`
+    tags: product_tags,
+    
+    // The type of the `ratings` field is a list of integers
+    ratings: { type: list, element: int },
+  }
+}
+```
+
+### Defining an enumeration
+
+Enumerations are defined using `valid_values` constraint which contains only symbol values.
+
+```
+$ion_schema_2_0
+type::{
+   name: fruits,
+   valid_values: [apple, banana, strawberry]
+}
+```
+
+### Special considerations for inline types
+
+Inline types in Ion Schema do not have a corresponding name for it. Here’s how the inline types are defined in ISL:
+
+```
+<INLINE_TYPE_DEFINITION> ::= { <CONSTRAINT>... }
+```
+
+For more information on inline types in Ion Schema,
+see [Type Definitions](https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#type-definitions).
+
+Since these inline types do not have name attached to it, code generator interprets the name of this inline type based
```

Comment:
The grammatical number does not match here. Try one of these instead.

> Since an inline type does not have a name attached to it...

or

> Since these inline types do not have names attached to them...